File: iris-crypto/iris-crypto-rsa/src/main/java/city/smartb/iris/crypto/rsa/signer/Signer.java
package city.smartb.iris.crypto.rsa.signer;

import com.nimbusds.jose.JWSAlgorithm;

import java.security.GeneralSecurityException;
import java.security.interfaces.RSAPrivateKey;

public interface Signer {

    static RS256Signer rs256Signer(RSAPrivateKey rsaPrivateKey) {
        return new RS256Signer(rsaPrivateKey);
    }

    JWSAlgorithm getAlgorithm();
    String getTerm();

    byte[] sign(byte[] content) throws GeneralSecurityException;


}

File: iris-crypto/iris-crypto-rsa-key/src/main/kotlin/city/smartb/iris/crypto/rsa/signer/Signer.kt
package city.smartb.iris.crypto.rsa.signer

import com.nimbusds.jose.JWSAlgorithm
import java.security.GeneralSecurityException
import java.security.interfaces.RSAPrivateKey

interface Signer {
    val algorithm: JWSAlgorithm?
    val term: String?

    @Throws(GeneralSecurityException::class)
    fun sign(content: ByteArray): ByteArray

    companion object {
        fun rs256Signer(rsaPrivateKey: RSAPrivateKey): RS256Signer {
            return RS256Signer(rsaPrivateKey)
        }
    }
}
